code/modules/reagents/reagent_containers/medspray.dm

```diff
@@ -91,3 +91,15 @@
 	name = "sterilizer spray"
 	desc = "Spray bottle loaded with non-toxic sterilizer. Useful in preparation for surgery."
 	list_reagents = list("sterilizine" = 60)
+/obj/item/reagent_containers/medspray/changed_name(newname)
```

Comment:
then add a listener and have this just be handled in update_icon()
`AddComponent(/datum/component/redirect, list(COMSIG_RENAMED = CALLBACK(src, .proc/update_icon)))`